staging/src/k8s.io/apiextensions-apiserver/pkg/controller/openapi/controller.go

```diff
@@ -77,16 +78,50 @@ func NewController(crdInformer informers.CustomResourceDefinitionInformer) *Cont
 	return c
 }
 
+func (c *Controller) setClusterCrdSpecs(clusterName, crdName string, newSpecs map[string]*spec.Swagger) {
```

Comment:
We didn't really talk about this before, but in the long run the "do this dynamically, not as part of a controller" is probably going to be important.  openapi/crd generation is expensive, so doing on a controller means that CPU and memory scale O(crds), when we really want them to scale O(active_clusters).

Reply:
I completely agree. That's probably one of the first big issue that we'll have to tackle from now on. This PR is clearly a half-step (even less than that of course). Everything regarding OpenAPI and resource discovery is managed through controllers currently (a number of controllers highly coupled with the corresponding http handlers). I tried to accommodate with this as much as possible when doing differently would have meant too much refactoring. This is why I did this for the prototype first step.
But I completely agree with the dynamic approach, though I'm not sure I completely foresee the how wide the refactoring impact is.